src/commandHandlers/roles.ts

```diff
@@ -1,35 +1,94 @@
 import { MessageEmbed } from 'discord.js';
 
 import { getRoles } from './guild.service';
-import config from '../config';
+import config, { selfAssignableRoles } from '../config';
 
 /**
  * This command lists the available roles on the discord server
  */
 export const command = async (arg: [string, string], embed: MessageEmbed) => {
   const roles = await getRoles();
-  const rolesList = roles
-    .filter((r) => !r.name.includes('everyone')) // Filter the default role everyone has access to
-    .map((discordRole) => {
-      const role = Object.values(config.ROLE).find(
-        (r) => r.name === discordRole.name
-      );
-      const roleDescription = role ? `- ${role.description}` : '';
-
-      return `\n• ${discordRole.toString()} ${roleDescription}`;
-    });
+  const specificRole = arg[1];
 
-  embed.setTitle('Available Roles')
-    .setDescription(`Here is the list of roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!
-        ${rolesList}
+  if (specificRole) {
+    if (arg[1] === '-a' || arg[1] === 'all') {
+      const describedRoles: string[] = [];
+      const rolesList: string[] = [];
+      roles.forEach((role) => {
+        if (role.name.includes('everyone')) {
+          return;
+        }
+        const roleConfig = Object.values(config.ROLE).find(
+          (r) => r.name === role.name
+        );
+        if (roleConfig && roleConfig.description) {
+          describedRoles.push(
+            `\n${role.toString()} - ${roleConfig.description}`
+          );
+        } else {
+          rolesList.push(` ${role.toString()}`);
+        }
+      });
+      embed.setTitle('Available Roles')
+        .setDescription(`Here is the list of all the roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!\n
+        ${rolesList.toString()}
+        ${describedRoles.toString()}\n
         Example of usage:
```

Comment:
Since this is a usage example of another command, maybe we should rename this to the same text we have for the Assignable roles option: "Assigning roles to yourself:".
What do you think @Vyvy-vi ?